### DISCUSSION TITLE
Setting up a bulletin main page, load error

### DISCUSSION THREAD

**Discussion by marralser** [NuSkooler/enigma-bbs#568] [Q&A]
Hi folks,

I have been experimenting with the wunderful Enigma 1/2 BBS system, but I'm totally new to hjson and the concept, so this proved  
 not totally trivial. But part of doing this is to learn, obviously :-).  I have the following question. I have been trying to set up a Bulletin menu, to be acessed from the main menu, simple idea, that a command, e.g. B1...B10, opens a respective ANSI graph. My code looks like this

```
{
    menus: {
        bulletinsMainMenu: {
            desc: Bulletins main menu
            art: BULLETINS
            prompt: menuCommand
            config: {
                interrupt: realtime
            }
            submit: [
                {
                    value: { command: "G" }
                    action: @menu:fullLogoffSequence
                }
                {
                    value: { command: "Q" }
                    action: @systemMethod:prevMenu
                }
                {
                    value: { command: "BU1" }
                    action: @menu:loadBulletin1
                }
                {
                    value: { command: "BU2" }
                    action: @menu:loadBulletin2
 }
 ]
        }
    }


        loadBulletin1: {
            desc: Bulletin 1
            art: BULLETIN1
            config: { pause: true }
        }
        loadBulletin2: {
            desc: Bulletin 2
            art: BULLETIN2
            config: { pause: true }
        }
}
```

The page loads correctly from the main menu, commends G and Q work, but not the BU commands. The error message inicates, the menu entry is not found 
"EnigError: Object does not exist: No menu entry for "loadBulletin1"
        at Object.DoesNotExist (/home/markus/enigma-bbs/core/enig_error.js:34:9)
        at locateMenuConfig (/home/markus/enigma-bbs/core/menu_util.js:32:40)
        at nextTask (/home/markus/enigma-bbs/node_modules/async/dist/async.js:5789:27)
        at Object.waterfall (/home/markus/enigma-bbs/node_modules/async/dist/async.js:5800:9)
        at Object.awaitable [as waterfall] (/home/markus/enigma-bbs/node_modules/async/dist/async.js:211:32)
        at getMenuConfig (/home/markus/enigma-bbs/core/menu_util.js:24:11)
        at getMenuConfiguration (/home/markus/enigma-bbs/core/menu_util.js:63:17)
        at nextTask (/home/markus/enigma-bbs/node_modules/async/dist/async.js:5789:27)
        at Object.waterfall (/home/markus/enigma-bbs/node_modules/async/dist/async.js:5800:9)
        at Object.awaitable [as waterfall] (/home/markus/enigma-bbs/node_modules/async/dist/async.js:211:32)"

I guess its a simple syntax mistake, but I can't spot it. Any idea highly appreciated!

Markus

**Comment by NuSkooler** [comment]
Hi @marralser! I took a look at the fragment here, it looks like the two `loadBulletin1` and `loadBulletin2` menus are not in the `menus` block.

I suggest using a editor that understands HJSON -- plugins exist for many editors if it's not built right in. This will let you see the alignment properly.

Below is an adjusted fragment; note the placement:
```
{
    menus: {
        bulletinsMainMenu: {
            desc: Bulletins main menu
            art: BULLETINS
            prompt: menuCommand
            config: {
                interrupt: realtime
            }
            submit: [
                {
                    value: { command: "G" }
                    action: @menu:fullLogoffSequence
                }
                {
                    value: { command: "Q" }
                    action: @systemMethod:prevMenu
                }
                {
                    value: { command: "BU1" }
                    action: @menu:loadBulletin1
                }
                {
                    value: { command: "BU2" }
                    action: @menu:loadBulletin2
                }
            ]
        }

        loadBulletin1: {
            desc: Bulletin 1
            art: BULLETIN1
            config: { pause: true }
        }
        loadBulletin2: {
            desc: Bulletin 2
            art: BULLETIN2
            config: { pause: true }
        }
    }
}
```

**Reply by marralser** [reply]
Ha thanks, that fixed it indeed, so it was syntax after all. Will take on your suggestion and use an editor that speaks hjson.
